plugins/woocommerce/src/Admin/API/Reports/Products/Stats/Segmenter.php

```diff
@@ -165,7 +165,7 @@ protected function get_segments( $type, $query_params, $table_name ) {
 
 			$segments = $this->get_product_related_segments( $type, $segmenting_selections, $segmenting_from, $segmenting_where, $segmenting_groupby, $segmenting_dimension_name, $table_name, $query_params, $unique_orders_table );
 		} elseif ( 'variation' === $this->query_args['segmentby'] ) {
-			if ( ! isset( $this->query_args['product_includes'] ) || count( $this->query_args['product_includes'] ) !== 1 ) {
+			if ( ! isset( $this->query_args['product_includes'] ) || ! is_array( count( $this->query_args['product_includes'] ) ) || count( $this->query_args['product_includes'] ) !== 1 ) {
```

Comment:
@asumaran it appears this change actually broke our Analytics view when comparing variation products within **Analytics > Products** and then selecting a variable product.
I assume this wasn't suppose to be wrapped in a `count` method?

Reply:
I created a PR with a fix: https://github.com/woocommerce/woocommerce/pull/45939

Reply:
@louwie17 I'll give a look at the PR.